pkg/client/restclient/config.go

```diff
@@ -283,11 +303,16 @@ func InClusterConfig() (*Config, error) {
 // Note: the Insecure flag is ignored when testing for this value, so MITM attacks are
 // still possible.
```

Comment:
Add to the comment that *any* non HTTPs host will return false.  

Also, what are the security implications of this?  I don't know that we should allow an admin to mix security types - I think we should return an error much earlier in the config process that guards against that (must be all http or all https)